### PR TITLE
feat: add spinner to code submission button

### PR DESCRIPTION
**Pull request by danielgent**
This duplicates the spinner functionality from the login2 button to here

~*BUT* we're not actually running React on the client :smile: , so I can add `isLoading` as a prop to the button~

Lol, not my greatest work, but this actually works :see_no_evil: 

[Screencast from 2024-05-23 10-54-25.webm](https://github.com/sesamyab/auth/assets/8496063/dc5d44e7-7ce3-44b1-93b8-ce9628cd5695)

### CODE REVIEW

File: src/utils/components/Button.tsx [hunk 1]
```diff
@@ -33,9 +38,22 @@ const Button: FC<Props> = ({
...
+      flex items-center justify-center space-x-2
+      ${isLoading ? "opacity-0" : ""}
+    `}
+      >
+        {children}
+      </span>
+      {isLoading && (
+        <div className="absolute left-0 top-0 flex h-full w-full items-center justify-center">
+          <Spinner size="medium" />
+        </div>
```
Comment (danielgent) [inline src/utils/components/Button.tsx:55]: copied from login2

File: src/utils/components/Layout.tsx [hunk 2]
```diff
@@ -130,6 +131,34 @@ const Layout: FC<LayoutProps> = ({ title, children, vendorSettings }) => {
...
+              if (code.length >= 6) {
+                console.log("Valid code");
+
+                document.getElementById("initial-btn").classList.add("hidden");
+                document
+                  .getElementById("loading-btn")
+                  .classList.remove("hidden");
+              }
+            };
+          }
```
Comment (danielgent) [inline src/utils/components/Layout.tsx:159]: :smile: Prettier even works (although not typescript)

File: src/utils/components/Spinner.tsx
Comment (danielgent) [inline src/utils/components/Spinner.tsx:1]: cloned from login2

File: src/utils/components/LoginEnterCodePage.tsx [hunk 3]
```diff
@@ -72,10 +72,14 @@ const LoginEnterCodePage: FC<Props> = ({
...
             required
+            id="code-input"
           />
           {error && <ErrorMessage>{error}</ErrorMessage>}
           <div class="text-center sm:mt-2">
-            <Button>{i18next.t("validate_code")}</Button>
+            <Button id="initial-btn"> {i18next.t("validate_code")}</Button>
+            <Button isLoading className="hidden" id="loading-btn">
+              {i18next.t("validate_code")}
+            </Button>
```
Comment (danielgent) [inline src/utils/components/LoginEnterCodePage.tsx:82]: :smile:

File: src/utils/components/Layout.tsx [hunk 4]
```diff
@@ -130,6 +131,24 @@ const Layout: FC<LayoutProps> = ({ title, children, vendorSettings }) => {
           </div>
         </div>
       </body>
+      {html`
```
Comment (danielgent) [inline src/utils/components/Layout.tsx:134]: JavaScript should go in the footer so the DOM is ready (I had flashbacks of developing a decade ago)

File: src/utils/components/Layout.tsx [hunk 5]
```diff
@@ -130,6 +131,24 @@ const Layout: FC<LayoutProps> = ({ title, children, vendorSettings }) => {
...
+        <script>
+          const initialBtn = document.getElementById("initial-btn");
+
+          if (initialBtn) {
+            document.getElementById("initial-btn").onclick = function () {
+              const code = document.getElementById("code-input").value;
+
+              if (code.length >= 6) {
```
Comment (danielgent) [inline src/utils/components/Layout.tsx:142]: We need to recreate this minLength validation here

**Review by markusahlstrand — APPROVED** [review]
Looks good. Might be interesting to see if we could use the client components from hono here.

**Pull request merged**